src/main/java/org/mifos/ops/zeebe/camel/routes/OperationsRouteBuilder.java

```diff
@@ -39,6 +41,67 @@
     @Override
     public void configure() {
 
+        /**
+         * Cancellation of the process by variable name and value
+         *
+         * sample request: {
+         * 	"key": "initiatorFspId",
+         * 	"value": "\"ibank-usa\""
+         * }
+         *
+         * sample response: {
+         *   "cancellationSuccessful": 0,
+         *   "cancellationFailed": 1,
+         *   "success": [],
+         *   "failed": [
+         *     2251799813686414
+         *   ]
+         * }
+         *
+         */
+        from("rest:POST:/channel/workflow/cancelbyvalue")
+                .id("cancel-workflow")
+                .log(LoggingLevel.INFO, "## Cancelling the process by matching variable name and value")
+                .process(exchange -> {
+
+                    JSONObject object = new JSONObject(exchange.getIn().getBody(String.class));
+                    String key = object.getString("key");
+                    Object value = object.get("value");
+
+                    TermsAggregationBuilder definitionNameAggregation = AggregationBuilders.terms("1")
+                            .field("value.processDefinitionKey")
+                            .size(5);
+
+                    BoolQueryBuilder query = QueryBuilders.boolQuery()
+                            .filter(QueryBuilders.matchPhraseQuery("value.name", key))
+                            .filter(QueryBuilders.matchPhraseQuery("value.value", value));
+
+                    SearchSourceBuilder builder = new SearchSourceBuilder().aggregation(definitionNameAggregation)
+                            .query(query);
+
+                    SearchRequest searchRequest =
+                            new SearchRequest().indices("zeebe-*").source(builder);
+
+                    SearchResponse response = esClient.search(searchRequest, RequestOptions.DEFAULT);
+
+                    String r = response.toString();
+                    JSONObject res = new JSONObject(r);
+                    JSONArray processDefinitionKeysObject = res.getJSONObject("aggregations").getJSONObject("lterms#1")
+                            .getJSONArray("buckets");
+
+                    JSONArray processIds = new JSONArray();
+
+                    processDefinitionKeysObject.forEach(elm -> {
+                        long processId = ((JSONObject) elm).getLong("key");
+                        processIds.put(processId);
+                    });
+
+                    JSONObject responseToBeReturned = cancelProcess(processIds);
+
+                    exchange.getMessage().setBody(responseToBeReturned.toString());
+
+                });
```

Comment:
This is fine. Are we also checking if the instance is in active state before sending it to be cancelled?